infrastructure/network/addressmanager/addressmanager.go

```diff
@@ -5,1264 +5,254 @@
-		// Tried and referenceCount are implicit in the rest of the structure
-		// and will be worked out from context on unserialisation.
-		peersState.Addresses[i] = serializedAddress
-		i++
-	}
-
-	peersState.SubnetworkNewAddressBucketArrays = make(map[string]*serializedNewAddressBucketArray)
-	for subnetworkID := range am.subnetworkNewAddressBucketArrays {
-		subnetworkIDStr := subnetworkID.String()
-		peersState.SubnetworkNewAddressBucketArrays[subnetworkIDStr] = &serializedNewAddressBucketArray{}
-
-		for i := range am.subnetworkNewAddressBucketArrays[subnetworkID] {
-			peersState.SubnetworkNewAddressBucketArrays[subnetworkIDStr][i] = make([]AddressKey, len(am.subnetworkNewAddressBucketArrays[subnetworkID][i]))
-			j := 0
-			for k := range am.subnetworkNewAddressBucketArrays[subnetworkID][i] {
-				peersState.SubnetworkNewAddressBucketArrays[subnetworkIDStr][i][j] = k
-				j++
-			}
-		}
-	}
-
-	for i := range am.fullNodeNewAddressBucketArray {
-		peersState.FullNodeNewAddressBucketArray[i] = make([]AddressKey, len(am.fullNodeNewAddressBucketArray[i]))
-		j := 0
-		for k := range am.fullNodeNewAddressBucketArray[i] {
-			peersState.FullNodeNewAddressBucketArray[i][j] = k
-			j++
+	for _, address := range addresses {
+		if !am.IsRoutable(address) {
+			continue
 		}
-	}
-
-	peersState.SubnetworkTriedAddressBucketArrays = make(map[string]*serializedTriedAddressBucketArray)
-	for subnetworkID := range am.subnetworkTriedAddresBucketArrays {
-		subnetworkIDStr := subnetworkID.String()
-		peersState.SubnetworkTriedAddressBucketArrays[subnetworkIDStr] = &serializedTriedAddressBucketArray{}
 
-		for i := range am.subnetworkTriedAddresBucketArrays[subnetworkID] {
-			peersState.SubnetworkTriedAddressBucketArrays[subnetworkIDStr][i] = make([]AddressKey, len(am.subnetworkTriedAddresBucketArrays[subnetworkID][i]))
-			j := 0
-			for _, knownAddress := range am.subnetworkTriedAddresBucketArrays[subnetworkID][i] {
-				peersState.SubnetworkTriedAddressBucketArrays[subnetworkIDStr][i][j] = NetAddressKey(knownAddress.netAddress)
-				j++
+		key := netAddressKey(address)
+		if _, ok := am.addresses[key]; !ok {
+			am.addresses[key] = &netAddressWrapper{
+				netAddress: address,
 			}
 		}
 	}
-
-	for i := range am.fullNodeTriedAddressBucketArray {
-		peersState.FullNodeTriedAddressBucketArray[i] = make([]AddressKey, len(am.fullNodeTriedAddressBucketArray[i]))
-		j := 0
-		for _, knownAddress := range am.fullNodeTriedAddressBucketArray[i] {
-			peersState.FullNodeTriedAddressBucketArray[i][j] = NetAddressKey(knownAddress.netAddress)
-			j++
-		}
-	}
-
-	return peersState, nil
 }
 
-// loadPeers loads the known address from the database. If missing,
-// just don't load anything and start fresh.
-func (am *AddressManager) loadPeers() error {
+// AddLocalAddresses adds local netAddresses to the address manager
+func (am *AddressManager) AddLocalAddresses(addresses ...*appmessage.NetAddress) {
```

Comment:
During the refactor, we have lost the functionality that was calling `AddLocalAddress` (under addressManager.initListeners), this functionality should be restored.